tests/testmplex.nim

```diff
@@ -140,6 +144,48 @@ suite "Mplex":
       let mplexDial = newMplex(conn)
       let stream  = await mplexDial.newStream()
       await stream.writeLp("Hello from stream!")
+      await lock or timeout
```

Comment:
I would actually use `wait` from `nim-chronos` here instead, if it timed out then wait throws an exception, `nim-unittest` has an `expect` clause to check for expected exceptions and we can cleanup in the finally clause of our `try`.

Reply:
> > They were basically never waiting reliably due to the nature of the async scheduler, you would barely notice if the messages were small because the operation probably was sequential once the async callback triggered.
> 
> Can you point me to an example? I'm pretty conscious about properly awaiting things where it matters.

For example doing like this:
https://github.com/sinkingsugar/nim-libp2p/commit/d6813775352e4bbdf5b5cea6de50d72a85200197
Btw, there is nothing really wrong, it's just error prone with delays (and sending lot of data gets split into different async scheduler iterations likely?)
Basically anything that happens inside `handleMplexListen` is not guaranteed to happen, and so that `lock` (probably bad name) waits or timeouts for that.

Btw nice catch on closing transports, that is likely why my big buffer was getting pushed into the next test 😸

Reply:
Btw I'm thinking to do the transport closing as part of https://github.com/status-im/nim-libp2p/issues/75

Reply:
SGTM, lets move cleanup of things to a central location.